extensions/2.0/Vendor/MSFT_RigidBodies/schema/node.MSFT_RigidBodies.rigidBody.schema.json

```diff
@@ -0,0 +1,69 @@
+{
+    "$schema" : "http://json-schema.org/draft-04/schema",
+    "title" : "MSFT_RigidBodies glTF extension",
+    "type" : "object",
+    "description" : "Parameters describing how a node's transform should be driven by physics simulation.",
+    "allOf" : [ { "$ref" : "glTFChildOfRootProperty.schema.json" } ],
+    "properties" : {
+        "isKinematic" : {
```

Comment:
Thoughts on adding a static rigid body, which is a kinematic body that must never move during simulation? A rigid body would then have a `type`  with three possible values: `"dynamic"` (default), `"kinematic"`, or `"static"`

Reply:
I don't think it's a good idea to have those three values. From the perspective of a physics simulation, there's only two kinds of bodies:

* Static bodies, which do not move
* Dynamic bodies, which have mass and velocity

Kinematic bodies are a special-case of dynamic bodies, where, when solving the equations of motion, they have zero gravity factor, zero damping and infinite mass/inertia.

So, this spec philosophically mirrors that. A "naked" collider is treated as static; then, making the node a `rigidBody` imbues that collider with a mass and velocity and then, finally, making it kinematic gives it infinite effective virtual mass.

I'd really resist adding "staticness" to the "rigidBody" object - it makes the `linearVelocity` and `angularVelocity` properties meaningless - that whole configuration space is  has no effect on a static body.

Reply:
> So, this spec philosophically mirrors that. A "naked" collider is treated as static;

Makes sense to me now.

Reply:
By this logic, it seems strange to allow a dynamic body, which by your definition has mass, to be specified as kinematic, which discards the mass. It's similar to a static body discarding the velocity.

My logic is that bodies should be primarily about combining together colliders, and secondarily about deciding how those colliders move around. It can be helpful to group together multiple static colliders together which would otherwise be "naked". It can also be helpful to group together multiple trigger colliders so that you can have a more complicated trigger volume as one volume instead of multiple.

As an example use case, in my company's game [The Mirror](https://www.themirror.space/), we allow the user to alter the imported data in-game as they wish, including by taking static colliders and making them rigid, so the ability to group together static colliders in one static body is very helpful for our use cases.

I don't have much else to say on this - I think I have made my point clear in several places. I've tried to avoid bringing up my specific uses to try and make a general case but at this point I feel like I have to.

Reply:
@aaronfranke has created a great table showcasing how a `type` property might map to various engines: https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_physics_body#physics-types

Personally, I think a `type` property would be easier to work with and more clear than the `isKinematic` flag.